### PR TITLE
Fixes for small configuration problems with environment module dependency resolvers.

### LINKED ISSUE #4452: Activating the "modules" dependency resolver prevent a Galaxy instance to start properly
If you try to uncomment the "**modules**" dependency resolver in the **dependency_resolvers_conf.xml** file and try to start the Galaxy instance then it will fail with the following errors:

```
Traceback (most recent call last):
  File "./scripts/manage_tool_dependencies.py", line 87, in <module>
    main()
  File "./scripts/manage_tool_dependencies.py", line 26, in main
    action_func(args)
  File "./scripts/manage_tool_dependencies.py", line 33, in _init_if_needed
    _build_dependency_manager_no_config(kwargs)
  File "./scripts/manage_tool_dependencies.py", line 76, in _build_dependency_manager_no_config
    dependency_manager = DependencyManager( **dependency_manager_kwds )
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/tools/deps/__init__.py", line 84, in __init__
    self.dependency_resolvers = self.__build_dependency_resolvers( conf_file )
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/tools/deps/__init__.py", line 195, in __build_dependency_resolvers
    return self.__parse_resolver_conf_xml( plugin_source )
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/tools/deps/__init__.py", line 210, in __parse_resolver_conf_xml
    return plugin_config.load_plugins( self.resolver_classes, plugin_source, extra_kwds )
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/util/plugin_config.py", line 36, in load_plugins
    return __load_plugins_from_element(plugins_dict, source, extra_kwds)
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/util/plugin_config.py", line 48, in __load_plugins_from_element
    plugin = plugins_dict[ plugin_type ]( **plugin_kwds )
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/tools/deps/resolvers/modules.py", line 47, in __init__
    self.module_checker = AvailModuleChecker(self, self.modulepath, prefetch, self.default_indicator)
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/tools/deps/resolvers/modules.py", line 118, in __init__
    for module in self.__modules():
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/tools/deps/resolvers/modules.py", line 137, in __modules
    raw_output = self.__module_avail_output().decode("utf-8")
  File "/home/arbernard/galaxy_project/galaxy/lib/galaxy/tools/deps/resolvers/modules.py", line 156, in __module_avail_output
    return Popen(avail_command, stderr=PIPE, env={'MODULEPATH': self.modulepath}).communicate()[1]
  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
    raise child_exception
OSError: [Errno 2] Aucun fichier ou dossier de ce type
```

The message in the log file is a bit more explicit:

> IOError: [Errno 2] Aucun fichier ou dossier de ce type: ./config/environment_modules_mapping.yml.sample'

There is indeed no such file in the config folder of recent Galaxy version.
I didn't found any documentation about this file yet.

### PR DESCRIPTION
**Pull request by jmchilton**
None of this broke the functionality but it made it harder to get going with and obscured errors.

Fixes #4452 I think - thanks @arbernard.

### CODE REVIEW

File: lib/galaxy/tools/deps/resolvers/modules.py [hunk 1]
```diff
@@ -154,7 +155,10 @@ def __modules(self):
                 yield module_name, module_version
 
     def __module_avail_output(self):
-        avail_command = [self.module_dependency_resolver.modulecmd, 'sh', 'avail']
+        modulecmd = self.module_dependency_resolver.modulecmd
+        if not exists(modulecmd):
```
Comment (nsoranzo) [inline lib/galaxy/tools/deps/resolvers/modules.py:159]: In some module implementations, the `module` command is actually a bash function. E.g. on my cluster:
```
$ which module 
/usr/bin/which: no module in ...
$ declare -f module
module () 
{ 
    BRIGHT='\e[1m';
...
```

Comment (jmchilton) [reply lib/galaxy/tools/deps/resolvers/modules.py:159]: Good point - I'm marking this WIP.

**Comment by arbernard** [comment]
**LMOD** environment module system (https://www.tacc.utexas.edu/research-development/tacc-projects/lmod) has a "module" command but it's indeed a bash function:

```
~$ type module
module est une fonction
module () 
{ 
    eval $($LMOD_CMD bash "$@") && eval $(${LMOD_SETTARG_CMD:-:} -s sh)
}
```
If the Galaxy admin set the **modulecmd** attribute of the **modules** resolver to "module" it will fail because suprocess.Popen is expecting an executable.

There is also a few other differences between the original module system and LMOD. If i'm not mistaken, the output of the "module avail" command is slighty different between the two, some environment variables are different and I didn't saw a prefetch option in LMOD yet).

Maybe a distinct resolver for LMOD should be better and less confusing for end "users" ? I've started to develop and test one last week if needed.

**Comment by mvdbeek** [comment]
I think this become obsolete with https://github.com/galaxyproject/galaxy/pull/4489, right ? if not feel free to re-open.

**Pull request closed without merging**